### PR TITLE
Rotation functionality

### PR DESCRIPTION
**Pull request by jeffreypaul15**
The easiest way rotation could be implemented in Pyvista without having to handle the camera's coordinates and line of sight.
This requires the class to keep track of all the meshes used because once `add_mesh` is called, there's no way to get a reference to it
The initial plot with a rotation of 30*u.deg is 
image

### CODE REVIEW

**Comment by nabobalis** [comment]
This needs rebasing/squash merging.

**Comment by nabobalis** [comment]
While we are here: 

Can you also take https://github.com/sunpy/sunpy/pull/4178/files?file-filters%5B%5D=.py#diff-85933aa74a2d66c3e4dcdf7a9ad8397f5a7971080d34ef1108296a7c6b69e7e3R149 
and add them to the docs/conf.py file?

**Comment by jeffreypaul15** [comment]
> While we are here:
> 
> Can you also take https://github.com/sunpy/sunpy/pull/4178/files?file-filters%5B%5D=.py#diff-85933aa74a2d66c3e4dcdf7a9ad8397f5a7971080d34ef1108296a7c6b69e7e3R149
> and add them to the docs/conf.py file?

Sure

File: examples/skip_rotating_maps.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
+"""
```
Comment (nabobalis) [inline examples/skip_rotating_maps.py:1]: What is the difference between this example and the first one?

Comment (jeffreypaul15) [reply examples/skip_rotating_maps.py:1]: `plotter.rotate(angle=30 * u.deg)` is all I suppose.

Comment (nabobalis) [reply examples/skip_rotating_maps.py:1]: Can you just add that to the first example?

File: sunkit_pyvista/plotter.py [hunk 2]
```diff
@@ -135,8 +139,14 @@ def plot_solar_axis(self, length=2.5, arrow_kwargs={}, **kwargs):
...
-                         **defaults)
-        self.plotter.add_mesh(arrow, **kwargs)
+        solar_axis = pv.Arrow(start=(0, 0, -length / 2),
+                              direction=(0, 0, length),
+                              scale='auto',
+                              **defaults)
+        self.plotter.add_mesh(solar_axis, **kwargs)
+        self.meshes.append(solar_axis)
+
+    def rotate(self, angle: u = 0.0 * u.deg):
```
Comment (nabobalis) [inline sunkit_pyvista/plotter.py:149]: This should have a docstring.

File: examples/skip_rotating_maps.py (hunk 1 above)
Comment (jeffreypaul15) [reply examples/skip_rotating_maps.py:1]: Yeah, I guess that makes more sense

File: sunkit_pyvista/plotter.py (hunk 2 above)
Comment (jeffreypaul15) [reply sunkit_pyvista/plotter.py:149]: I'll add that in

**Review by dstansby — CHANGES_REQUESTED** [review]
I don't think the `rotate()` method applies to 3D meshes in the same way it is handling for 2D maps. If we do want to add a rotate functionality, it should rotate the camera, and not the meshes. This can be done by setting the `plotter.camera_position` property. It's not very well documented, but https://github.com/pyvista/pyvista-support/issues/40#issuecomment-518738649 has an example. I think we should discuss exposing ways to set the camera position, instead of modifying all the meshes. Since there's not much other stuff in this PR, we could either close it, or you could leave the nitpicky stuff in here? (that should definitely get added!)

**Comment by jeffreypaul15** [comment]
> I don't think the `rotate()` method applies to 3D meshes in the same way it is handling for 2D maps. If we do want to add a rotate functionality, it should rotate the camera, and not the meshes. This can be done by setting the `plotter.camera_position` property. It's not very well documented, but [pyvista/pyvista-support#40 (comment)](https://github.com/pyvista/pyvista-support/issues/40#issuecomment-518738649) has an example. I think we should discuss exposing ways to set the camera position, instead of modifying all the meshes. Since there's not much other stuff in this PR, we could either close it, or you could leave the nitpicky stuff in here? (that should definitely get added!)

Ah yeah that does require some but if tinkering, I couldn't find any information regarding that. I'll close this PR, figure it out and open a new one. I assumed this would be an easier alternative, my apologies

**Comment by nabobalis** [comment]
Can the nitpicky go in a new PR?

**Pull request closed without merging**